Google.Api.Generator/RoslynUtils/RoslynExtensions.cs

```diff
@@ -61,6 +64,22 @@ private static T WithBody<T>(IEnumerable<object> code, Func<ArrowExpressionClaus
         public static MethodDeclarationSyntax WithBody(this MethodDeclarationSyntax method, params object[] code) =>
             WithBody(code, x => method.WithExpressionBody(x).WithSemicolonToken(s_semicolonToken), method.WithBody);
 
+        public static ObjectCreationExpressionSyntax WithInitializer(
+            this ObjectCreationExpressionSyntax obj, params (string propertyName, object code)[] inits)
+        {
+            if (!obj.ArgumentList.Arguments.Any())
+            {
+                // If calling the parameterless ctor, recreate the `new` expression without an argument list.
```

Comment:
Thank you! (I assume this means `new Foo { Bar = X }` rather than `new Foo() { Bar = X }`?)

Reply:
Yes, exactly.